sources/packages/backend/apps/api/src/route-controllers/student/student.institutions.controller.ts

```diff
@@ -0,0 +1,44 @@
+import { Body, Controller, Post } from "@nestjs/common";
+import { ApiTags } from "@nestjs/swagger";
+import { InstitutionService, StudentService } from "../../services";
+import { ClientTypeBaseRoute } from "../../types";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { AllowAuthorizedParty, UserToken } from "../../auth/decorators";
+import BaseController from "../BaseController";
+import {
+  StudentSearchAPIInDTO,
+  SearchStudentAPIOutDTO,
+} from "./models/student.dto";
+import { IInstitutionUserToken } from "../../auth";
+
+/**
+ * Student controller for institutions.
+ */
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@Controller("student")
+@ApiTags(`${ClientTypeBaseRoute.Institution}-student`)
+export class StudentInstitutionsController extends BaseController {
+  constructor(
+    private readonly studentService: StudentService,
+    private readonly institutionService: InstitutionService,
+  ) {
+    super();
+  }
+
+  /**
+   * Search students based on the search criteria.
+   * TODO add decorator to restrict to BC Public institutions.
+   * @param searchCriteria criteria to be used in the search.
+   * @returns searched student details.
+   */
+  @Post("search")
+  async searchStudents(
+    @UserToken() userToken: IInstitutionUserToken,
+    @Body() searchCriteria: StudentSearchAPIInDTO,
+  ): Promise<SearchStudentAPIOutDTO[]> {
+    return await this.studentService.searchStudentApplication(
```

Comment:
Maybe I am missing something but is the service result directly returned through the API?
The results must be converted to a DTO.

Reply:
Now I see that the service is returning the DTO directly. The services, unless they are controller services, do not return DTO directly. API DTOs are supposed to be a controller concern only, that is why we usually deal with them at controller or service controller only.